test/unit/input_convertors.py

```diff
@@ -34,8 +47,10 @@ def ensure_ndarray(X):
 def ensure_ndarray_2d(X):
     assert X is not None
     tmp = ensure_ndarray(X)
+    if len(tmp.shape) == 2:
+        return tmp
     if len(tmp.shape) != 1:
-        raise ValueError("Requires 1d array")
+        raise ValueError("Requires 1d or 2d array")
```

Comment:
Can you do this with an `if … elif … else` construct? Right now, the error message doesn't match the if statement unless you look carefully at the preceding `return` logic.

Reply:
Agreed, this is kind of weird right now.

Reply:
I moved this check before the other one which makes it more obvious (I think)